src/lib.rs

```diff
@@ -105,42 +97,200 @@ impl Default for FxHasher {
 impl FxHasher {
     #[inline]
     fn add_to_hash(&mut self, i: usize) {
-        self.hash = self.hash.rotate_left(5).bitxor(i).wrapping_mul(K);
+        self.hash = self.hash.wrapping_add(i).wrapping_mul(K);
     }
 }
 
 impl Hasher for FxHasher {
     #[inline]
-    fn write(&mut self, mut bytes: &[u8]) {
-        // Ensure all bytes will be consumed
-        const _: () = assert!(size_of::<usize>() <= size_of::<u64>());
-        // Ensure no bytes are discarded by casting to usize
-        const _: () = assert!(size_of::<u32>() <= size_of::<usize>());
-        // Copy the 1 word sized state to a local variable to ensure it
-        // is kept in a register.
-        // See: https://github.com/rust-lang/rustc-hash/pull/34
-        let mut state = self.clone();
-        while let Some(&usize_bytes) = take_first_chunk(&mut bytes) {
-            state.add_to_hash(usize::from_ne_bytes(usize_bytes));
-        }
-        if let Some(&u32_bytes) = take_first_chunk(&mut bytes) {
-            state.add_to_hash(u32::from_ne_bytes(u32_bytes) as usize);
-        }
-        if let Some(&u16_bytes) = take_first_chunk(&mut bytes) {
-            state.add_to_hash(u16::from_ne_bytes(u16_bytes) as usize);
-        }
-        if let Some(&[u8_byte]) = take_first_chunk(&mut bytes) {
-            state.add_to_hash(u8_byte as usize);
-        }
-        *self = state;
+    fn write(&mut self, bytes: &[u8]) {
+        self.write_u64(hash_bytes(bytes));
```

Comment:
It's surprising to me that the result of `hash_bytes` gets hashed. Judging from the last sentence of the comment on `hash_bytes` this is deliberate? If so, a brief comment about it would be useful here.

Reply:
`hash_bytes` only compresses a string into a single `u64`. It still needs to be mixed into the state, which is what `write_u64` does.